ui/components/multichain/pages/send/components/quote-card/hooks/useEthFeeData.tsx

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { Numeric } from '../../../../../../../../shared/modules/Numeric';
+import {
+  getConversionRate,
+  getGasFeeEstimates,
+  getNativeCurrency,
+} from '../../../../../../../ducks/metamask/metamask';
+import { EtherDenomination } from '../../../../../../../../shared/constants/common';
+import {
+  getCurrentCurrency,
+  checkNetworkAndAccountSupports1559,
+} from '../../../../../../../selectors/selectors';
+import { getUsedSwapsGasPrice } from '../../../../../../../ducks/swaps/swaps';
+import { formatCurrency } from '../../../../../../../helpers/utils/confirm-tx.util';
+import { toFixedNoTrailingZeros } from './utils';
+
+export default function useEthFeeData(gasLimit = 0) {
```

Comment:
Are you adding unit tests for this?

Reply:
Gonna just do unit tests in the e2e testing PR if that's cool